File: app/src/main/java/com/ddancn/keepaccount/adapter/RecordAdapter.java
package com.ddancn.keepaccount.adapter;

import androidx.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.ddancn.keepaccount.R;
import com.ddancn.keepaccount.constant.TypeEnum;
import com.ddancn.keepaccount.entity.Record;

import java.util.List;

/**
 * @author ddan.zhuang
 */
public class RecordAdapter extends BaseQuickAdapter<Record, BaseViewHolder> {

    public RecordAdapter(int layoutResId) {
        super(layoutResId);
    }

    public RecordAdapter(int layoutResId, @Nullable List<Record> data) {
        super(layoutResId, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, Record item) {
        helper.setText(R.id.tv_date, item.getDate())
                .setText(R.id.tv_type, item.getTypeName())
                .setText(R.id.tv_detail, item.getDetail())
                .setText(R.id.tv_money, String.valueOf(item.getMoney()))
                .setTextColor(R.id.tv_money, item.getType() == TypeEnum.IN.value()
                        ? mContext.getResources().getColor(R.color.colorPrimary)
                        : mContext.getResources().getColor(R.color.colorText));
    }

}

File: app/src/main/java/com/ddancn/keepaccount/adapter/RecordAdapter.kt
package com.ddancn.keepaccount.adapter

import androidx.annotation.LayoutRes
import com.chad.library.adapter.base.BaseSectionQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.ddancn.keepaccount.R
import com.ddancn.keepaccount.constant.TypeEnum
import com.ddancn.keepaccount.vo.RecordSection

/**
 * @author ddan.zhuang
 */
class RecordAdapter(@LayoutRes layoutResId: Int, @LayoutRes sectionHeadResId: Int, data: List<RecordSection>? = null) :
    BaseSectionQuickAdapter<RecordSection, BaseViewHolder>(layoutResId, sectionHeadResId, data) {

    override fun convert(helper: BaseViewHolder, item: RecordSection) {
        val record = item.t
        helper.setText(R.id.tv_category, record.categoryName)
            .setText(R.id.tv_detail, record.detail)
            .setText(R.id.tv_money, record.money.toString())
            .setTextColor(
                R.id.tv_money, if (record.type == TypeEnum.IN.value())
                    mContext.resources.getColor(R.color.colorPrimary, mContext.theme)
                else mContext.resources.getColor(R.color.colorText, mContext.theme)
            )
    }

    override fun convertHead(helper: BaseViewHolder, item: RecordSection) {
        helper.setText(R.id.tv_date, item.header)
    }
}
